aws-lambda-java-tests/src/test/resources/ddb/dynamo_ddb_stream_record.json

```diff
@@ -0,0 +1,35 @@
+{
+    "eventID": "c81e728d9d4c2f636f067f89cc14862c",
+    "eventName": "MODIFY",
+    "eventVersion": "1.1",
+    "eventSource": "aws:dynamodb",
+    "awsRegion": "eu-central-1",
+    "dynamodb": {
+        "Keys": {
+            "Id": {
+                "N": "101"
+            }
+        },
+        "NewImage": {
+            "Message": {
+                "S": "This item has changed"
+            },
+            "Id": {
+                "N": "101"
+            }
+        },
+        "OldImage": {
+            "Message": {
+                "S": "New item!"
+            },
+            "Id": {
+                "N": "101"
+            }
+        },
+        "ApproximateCreationDateTime": 1428537600,

```

Comment:
I think this is represented as a floating point number in this event, i.e.`1428537600.0`

Reply:
The already existing [record with `eventName=MODIFY` in `dynamo_event.json`](https://github.com/aws/aws-lambda-java-libs/blob/02e3cdc3caa6a4aee2fb4c3c221191994709e080/aws-lambda-java-tests/src/test/resources/ddb/dynamo_event.json#L34-L68) was reused.
According to [StreamRecord.ApproximateCreationDateTime - Amazon DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_StreamRecord.html#:~:text=ApproximateCreationDateTime) documentation, a `long` number should be considered:

> The approximate date and time when the stream record was created, in [UNIX epoch time](http://www.epochconverter.com/) format and rounded down to the closest second.
> Type: Timestamp

Also, no decimals are considered in the (_not edited_) [corresponding test assertion](https://github.com/aws/aws-lambda-java-libs/pull/455/files/02e3cdc3caa6a4aee2fb4c3c221191994709e080#diff-46ce264a2374bf6be6602d1f35ac3e89719b50bc7e11ed5acefc34e791a8eaffR185) using:
[java.util.Date.from](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#from-java.time.Instant-)([java.time.Instant.ofEpochSecond(long)](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#ofEpochSecond-long-)).

Reply:
Hi @smirnoal,

In AWS you might see a dotted number representation for `StreamRecord.ApproximateCreationDateTime`, but is its scientific (number in exponential) notation - observed sample:

```
ApproximateCreationDateTime=1.697900578E9
```

The `long` number representation for [UNIX epoch time](http://www.epochconverter.com/) `1697900578` - ISO8601: `2023-10-21T15:02:58Z`.

Reply:
I recently saw event like below from PROD
```
{
    'Records': [{
        'eventID': '5f342e95ccef0cf90ff71be3fc9c96e5',
        'eventName': 'INSERT',
        'eventVersion': '1.1',
        'eventSource': 'aws:dynamodb',
        'awsRegion': 'us-east-1',
        'dynamodb': {
            'ApproximateCreationDateTime': 1697219385.0,
            'Keys': {
                'date': {
                    'S': '20230101'
                },
                'playerId': {
                    'S': 'abcd'
                }
            },
            'NewImage': {
                'date': {
                    'S': '20230101'
                },
                'playerId': {
                    'S': 'abcd'
                }
            },
            'SequenceNumber': '3131009900000000046478534043',
            'SizeBytes': 48,
            'StreamViewType': 'NEW_AND_OLD_IMAGES'
        },
        'eventSourceARN': 'arn:aws:dynamodb:us-east-1:<accountid>:table/GameEvents/stream/2021-12-24T20:58:44.015'
    }]
}
```

Reply:
Updated in commit 232497ac.

Reply:
We have seen epoch format for  ApproximateCreationDateTime attribute in dynamodb event
"ApproximateCreationDateTime": 1698752983,

And dynamodb event via kinesis is in millisecond precision  
"ApproximateCreationDateTime":1698658039410,

Reply:
Hi @vaibhav170,

Thanks for the feedback.

Do you think it worth adding different tests for each and every possible and already observed `ApproximateCreationDateTime` value when AWS states that is [rounded down to the closest second](https://github.com/aws/aws-lambda-java-libs/pull/455#discussion_r1372401195)?

Reply:
i think second level precision test is fine for dynamodb stream event to align with documentation

Reply:
Updated in commit f92b0bc5.

Reply:
Hi @smirnoal,

Sorry to ask, but what is required for this PR to be approved&merged?

Thanks

Reply:
hey sorry was a bit busy with java21 launch. Will get to this next week.

Reply:
I think you will need to rebase it before merging: "This branch is out-of-date with the base branch"